kvui.py

```diff
@@ -386,6 +386,60 @@ def insert_text(self, substring, from_undo=False):
         return super(ConnectBarTextInput, self).insert_text(s, from_undo=from_undo)
 
 
+def is_command_input(string: str) -> bool:
+    return len(string) > 0 and string[0] in "/!"
+
+
+class CommandPromptTextInput(TextInput):
+    MAXIMUM_HISTORY_MESSAGES = 50
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self._command_history_index = -1
+        self._command_history: typing.List[str] = []
```

Comment:
you'll want a collections.deque for this, it's optimized  for such operations.

Reply:
Good recommendation, switched over, the `maxlen` also automatically handles the max size for me. Tested by temporarily setting the max size down to 5 locally and running up to the command buffer limit, and the history pointer correctly wouldn't go earlier.